src/iosMain/kotlin/com/airthings/lib/logging/platform/PlatformFileInputOutputImpl.kt

```diff
@@ -52,26 +52,21 @@ private typealias NSERROR_CPOINTER = CPointer<ObjCObjectVar<NSError?>>
 internal actual class PlatformFileInputOutputImpl : PlatformFileInputOutput {
     override val pathSeparator: Char = '/'
 
-    override suspend fun isDirectory(path: String): Boolean = memScoped {
-        val cValue = alloc<BooleanVar>()
-        cValue.value = true
-        NSFileManager.defaultManager.fileExistsAtPath(
-            path = path,
-            isDirectory = cValue.ptr,
-        )
-    }
-
     override suspend fun size(path: String): Long = nsErrorWrapper(0L) {
         sizeImpl(path)
     }
 
-    override suspend fun mkdirs(path: String): Boolean = nsErrorWrapper(false) {
-        NSFileManager.defaultManager.createDirectoryAtPath(
-            path = path,
-            withIntermediateDirectories = true,
-            attributes = null,
-            error = this,
-        )
+    override suspend fun mkdirs(path: String): Boolean {
+        nsErrorWrapper(false) {
+            NSFileManager.defaultManager.createDirectoryAtPath(
+                path = path,
+                withIntermediateDirectories = true,
+                attributes = null,
+                error = this,
```

Comment:
`error = this` 🤔 this is a pointer in NS world should either be `null` or `error:&error`

Reply:
In the Kotlin world, it's wrapped inside a `ObjCObjectVar` instance, please take a look at the reusable function `nsErrorWrapper()` found at the end of this file.

Basically, in Kotlin, you allocate a certain memory position and reserve it as a pointer location. Then the pointer location is provided to the caller as `this`.